pkg/collector/corechecks/servicediscovery/module/envs.go

```diff
@@ -61,55 +44,103 @@ func readEnvsFile(path string) ([]string, error) {
 // lrwx------ 1 foo foo 64 Aug 13 14:24 2 -> /dev/pts/6
 // lrwx------ 1 foo foo 64 Aug 13 14:24 3 -> '/memfd:dd_environ (deleted)'
 // ```
-func getInjectedEnvs(proc *process.Process) []string {
-	fdsPath := kernel.HostProc(strconv.Itoa(int(proc.Pid)), "fd")
-	entries, err := os.ReadDir(fdsPath)
+func getInjectionMeta(proc *process.Process) (*InjectedProcess, bool) {
+	path, found := findInjectorFile(proc)
+	if !found {
+		return nil, false
+	}
+	injectionMeta, err := extractInjectionMeta(path)
 	if err != nil {
-		return nil
+		log.Warnf("error extracting injected envs: %s", err)
```

Comment:
Nit: warn logs that start with `error` raise a question - why isn't it an error log?
small change:
```suggestion
		log.Warnf("failed extracting injected envs: %s", err)
```